artifactory/commands/transferfiles/fulltransfer.go

```diff
@@ -137,68 +137,77 @@ func (m *fullTransferPhase) transferFolder(params folderParams, logMsgPrefix str
-			_, err = pcDetails.producerConsumer.AddTaskWithError(folderHandler(folderParams{repoKey: params.repoKey, relativePath: newRelativePath}), pcDetails.errorsQueue.AddError)
-			if err != nil {
-				return err
-			}
-		case "file":
-			file := FileRepresentation{Repo: item.Repo, Path: item.Path, Name: item.Name}
-			delayed := delayHelper.delayUploadIfNecessary(file)
-			if delayed {
+
+		// Empty folder. Add it as candidate.
+		if paginationI == 0 && len(result.Results) == 0 {
+			curUploadChunk.appendUploadCandidate(FileRepresentation{
+				Repo: params.repoKey,
+				Path: path.Dir(params.relativePath),
+				Name: path.Base(params.relativePath),
+			})
+			break
+		}
+
+		for _, item := range result.Results {
+			if item.Name == "." {
 				continue
 			}
-			curUploadChunk.appendUploadCandidate(file)
-			if len(curUploadChunk.UploadCandidates) == uploadChunkSize {
-				err := uploadChunkWhenPossible(m.srcUpService, curUploadChunk, uploadTokensChan, errorChannel)
+			switch item.Type {
+			case "folder":
+				newRelativePath := item.Name
+				if params.relativePath != "." {
+					newRelativePath = path.Join(params.relativePath, newRelativePath)
+				}
+				folderHandler := m.createFolderFullTransferHandlerFunc(pcDetails, uploadTokensChan, delayHelper, errorChannel)
+				_, err = pcDetails.producerConsumer.AddTaskWithError(folderHandler(folderParams{repoKey: params.repoKey, relativePath: newRelativePath}), pcDetails.errorsQueue.AddError)
 				if err != nil {
-					log.Error(err)
+					return err
 				}
-				// Increase phase1 progress bar with the uploaded number of files.
-				if m.progressBar != nil {
-					err = m.progressBar.IncrementPhaseBy(m.phaseId, len(curUploadChunk.UploadCandidates))
+			case "file":
+				file := FileRepresentation{Repo: item.Repo, Path: item.Path, Name: item.Name}
+				delayed := delayHelper.delayUploadIfNecessary(file)
+				if delayed {
+					continue
+				}
+				curUploadChunk.appendUploadCandidate(file)
+				if len(curUploadChunk.UploadCandidates) == uploadChunkSize {
+					err := uploadChunkWhenPossible(m.srcUpService, curUploadChunk, uploadTokensChan, errorChannel)
 					if err != nil {
-						return err
+						log.Error(err)
+					}
+					// Increase phase1 progress bar with the uploaded number of files.
+					if m.progressBar != nil {
+						err = m.progressBar.IncrementPhaseBy(m.phaseId, len(curUploadChunk.UploadCandidates))
+						if err != nil {
+							return err
+						}
 					}
+					// Empty the uploaded chunk.
+					curUploadChunk.UploadCandidates = []FileRepresentation{}
 				}
-				// Empty the uploaded chunk.
-				curUploadChunk.UploadCandidates = []FileRepresentation{}
 			}
 		}
-	}
 
-	// Empty folder. Add it as candidate.
-	if len(result.Results) == 0 {
-		curUploadChunk.appendUploadCandidate(FileRepresentation{
-			Repo: params.repoKey,
-			Path: path.Dir(params.relativePath),
-			Name: path.Base(params.relativePath),
-		})
+		if len(result.Results) < aqlPaginationLimit {
```

Comment:
Shouldn't be `<=`?

Reply:
No, if `len(result.Results) == aqlPaginationLimit` we don't know if there are more results but they were not returned because of the limit. This will make this loop run once only.
When `len(result.Results) < aqlPaginationLimit` we know we can stop because the results weren't truncated by the limit.